libraries/ESP8266WiFi/src/ESP8266WiFiScan.cpp

```diff
@@ -102,6 +102,53 @@ int8_t ESP8266WiFiScanClass::scanNetworks(bool async, bool show_hidden) {
 
 }
 
+/**
+ * Start scan WiFi networks available
+ * @param async         run in async mode
+ * @param show_hidden   show hidden networks
+ * @param channel       scan only this channel (0 for all channels)
+ * @param ssid*         scan for only this ssid
+
+ * @return Number of discovered networks
+ */
+int8_t ESP8266WiFiScanClass::scanNetworks(bool async, bool show_hidden, uint8 channel, uint8* ssid) {
+    if(ESP8266WiFiScanClass::_scanStarted) {
+        return WIFI_SCAN_RUNNING;
+    }
+
+    ESP8266WiFiScanClass::_scanAsync = async;
+
+    WiFi.enableSTA(true);
+
+    int status = wifi_station_get_connect_status();
+    if(status != STATION_GOT_IP && status != STATION_IDLE) {
+        wifi_station_disconnect();
+    }
+
+    scanDelete();
+
+    struct scan_config config;
+    memset(&config, 0, sizeof(config));
+    config.ssid = ssid;

```

Comment:
The only difference between the original method and your overloaded one are these two lines:
 config.ssid = ssid;
 config.channel = channel;

where ssid and channel are new arguments.

Instead of duplicating the entire method, please just add the two arguments to the original method, and in the .h put set default values to keep the current behavior (i.e.: channel = 0 and ssid = nullptr, because of the memset).